internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -540,6 +550,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 
 	orig := xr.GetCompositionReference()
 	if err := r.composite.SelectComposition(ctx, xr); err != nil {
+		if kerrors.IsConflict(err) {
```

Comment:
is this error handling a direct impact of the changes in this PR, or just something that was always possible and you want to be more defensive about now?

Reply:
It turned out to be unrelated to the PR - I broke it out into a separate commit (https://github.com/crossplane/crossplane/pull/6321/commits/837870f797875791302adc9154b6084e720a3e3a).

I noticed it because at one point the realtime composition selection E2E test was failing, and I thought it might have been due to this. That didn't turn out to be the case though.